eclair-core/src/main/scala/fr/acinq/eclair/blockchain/watchdogs/BlockchainWatchdog.scala

```diff
@@ -77,10 +88,26 @@ object BlockchainWatchdog {
             Behaviors.same
           case CheckLatestHeaders(blockCount) =>
             val id = UUID.randomUUID()
-            context.spawn(HeadersOverDns(chainHash, blockCount), s"${HeadersOverDns.Source}-$blockCount-$id") ! HeadersOverDns.CheckLatestHeaders(context.self)
-            context.spawn(ExplorerApi(chainHash, blockCount, ExplorerApi.BlockstreamExplorer()), s"blockstream-$blockCount-$id") ! ExplorerApi.CheckLatestHeaders(context.self)
-            context.spawn(ExplorerApi(chainHash, blockCount, ExplorerApi.BlockcypherExplorer()), s"blockcypher-$blockCount-$id") ! ExplorerApi.CheckLatestHeaders(context.self)
-            context.spawn(ExplorerApi(chainHash, blockCount, ExplorerApi.MempoolSpaceExplorer()), s"mempool.space-$blockCount-$id") ! ExplorerApi.CheckLatestHeaders(context.self)
+            val chainHash = nodeParams.chainHash
+            val socksProxy_opt = nodeParams.socksProxy_opt
+            val sources = nodeParams.blockchainWatchdogSources
+            if (socksProxy_opt.isEmpty && sources.contains(HeadersOverDns.Source)) {
+              context.spawn(HeadersOverDns(chainHash, blockCount), s"${HeadersOverDns.Source}-$blockCount-$id") ! HeadersOverDns.CheckLatestHeaders(context.self)
+            } else {
+              context.log.warn(s"blockchain watchdog ${HeadersOverDns.Source} is disabled")
+            }
+            val explorers = Seq(ExplorerApi.BlockstreamExplorer(socksProxy_opt), ExplorerApi.BlockcypherExplorer(socksProxy_opt), ExplorerApi.MempoolSpaceExplorer(socksProxy_opt))
+              .foldLeft(Seq.empty[ExplorerApi.Explorer]) { (acc, w) =>
+                if (sources.contains(w.name)) {
+                  acc :+ w
+                } else {
+                  context.log.warn(s"blockchain watchdog ${w.name} is disabled")
+                  acc
+                }
+              }
+            explorers.foreach { explorer =>
+              context.spawn(ExplorerApi(chainHash, blockCount, explorer), s"${explorer.name}-$blockCount-$id") ! ExplorerApi.CheckLatestHeaders(context.self)
+            }
```

Comment:
There is no need to `fold`, you can directly apply your `foreach`:

```suggestion
Seq(ExplorerApi.BlockstreamExplorer(socksProxy_opt), ExplorerApi.BlockcypherExplorer(socksProxy_opt), ExplorerApi.MempoolSpaceExplorer(socksProxy_opt)).foreach { explorer =>
              if (sources.contains(explorer.name)) {
                context.spawn(ExplorerApi(chainHash, blockCount, explorer), s"${explorer.name}-$blockCount-$id") ! ExplorerApi.CheckLatestHeaders(context.self)
              }
            }
```

Similarly to `HeadersOverDns`, I'd like to move the log line advertising that a watchdog is disabled to the beginning of the `apply` method to ensure it's only logged once and not every 10 minutes.